scripts/IntentData.py

```diff
@@ -76,23 +76,27 @@ def addRawOutput(self, rawOutputs, labelsMap):
         """ Read the raw output and store all data from it - 
             channel outputs, context variables and jumpto definitions. """
         self._rawOutputs.append(rawOutputs)
-        if not isinstance(rawOutputs, tuple) or len(rawOutputs) < 1:
+        if not isinstance(rawOutputs, tuple) or sum([1 if x else 0 for x in rawOutputs]) == 0:
             eprintf('Warning: rawOutput does not contain any data: %s\n', rawOutputs)
         
-        for item in re.split('%%', rawOutputs[0]):
-            if not item: continue
-            if item.startswith(u'$'):
-                self.__handleVariableDefinition(item[1:])
-            elif item.startswith(u'B'):
-                self.__handleButtonDefinition(item[1:])
-            elif item.startswith(u'F'):
-                self.__handleFoldableDefinition(item[1:])
-            elif item.startswith(u':'):
-                self.__handleJumpToDefinition(item[1:], labelsMap)
-            else:
-                self.__handleChannelDefinition(item)
-
-        if len(rawOutputs) >= 3:
+        if len(rawOutputs) >= 1 and (isinstance(rawOutputs[0], str) or isinstance(rawOutputs[0], unicode)):
+            items = re.split('%%', rawOutputs[0])
+            self.__handleChannelDefinition(items[0])
+            for item_i in range(1, len(items)):
+                item = items[item_i]
+                if not item: continue
+                if item.startswith(u'$'):
+                    self.__handleVariableDefinition(item[1:])
+                elif item.startswith(u'B'):
+                    self.__handleButtonDefinition(item[1:])
+                elif item.startswith(u'F'):
+                    self.__handleFoldableDefinition(item[1:])
+                elif item.startswith(u':'):
+                    self.__handleJumpToDefinition(item[1:], labelsMap)
+                else:
+                    self.__handleChannelDefinition(item)
+
+        if len(rawOutputs) >= 2:
```

Comment:
Original behavior: process 'B', '$' modifiers for all chunks separated by %%.
New behavior: process them only for second and later chunks.
Original behavior caused simple text outputs that begin with B to not appear in the result xml file.

Reply:
Comment added into code.